portal/models/user.py

```diff
@@ -362,6 +362,8 @@ def is_registered(self):
         if not disjoint:
             return False
         else:
+            if not current_roles:
```

Comment:
No, I don't believe we want this clause.  The point is, if we created a non-registered user somehow, they should have one of the `PRE_REGISTERED_ROLES`.
Am I overlooking a legitimate use case @achen2401 ?